Code/Editor/Lib/Tests/test_AzAssetBrowserRequestHandler.cpp

```diff
@@ -0,0 +1,678 @@
+    // in response to a query about what kind of a component should be spawned
+    // for a given asset type.
+    class AssetTypeInfoBusHandlerMock : public AZ::AssetTypeInfoBus::Handler
+    {
+    public:
+        MOCK_CONST_METHOD0(GetComponentTypeId, AZ::Uuid());
+        MOCK_CONST_METHOD0(GetAssetType, AZ::Data::AssetType());
+        MOCK_CONST_METHOD0(GetAssetTypeDragAndDropCreationPriority, int());
+
+        void InstallHandlersFor(AZ::Uuid componentType, AZ::Data::AssetType assetType, int priority = 0)
+        {
+            ON_CALL(*this, GetComponentTypeId)
+                .WillByDefault(
+                    [componentType]()
+                    {
+                        return componentType;
+                    });
+
+            ON_CALL(*this, GetAssetType)
+                .WillByDefault(
+                    [assetType]()
+                    {
+                        return assetType;
+                    });
+
+            ON_CALL(*this, GetAssetTypeDragAndDropCreationPriority)
+                .WillByDefault(
+                    [priority]()
+                    {
+                        return priority;
+                    });
+
+            BusConnect(assetType);
+        }
+    };
+   
+    class MockComponentApplicationBusHandler : public AZ::ComponentApplicationBus::Handler
+    {
+        public:
+            MOCK_METHOD0(GetApplication, AZ::ComponentApplication*());
+            MOCK_METHOD1(RegisterComponentDescriptor, void(const AZ::ComponentDescriptor*));
+            MOCK_METHOD1(UnregisterComponentDescriptor, void(const AZ::ComponentDescriptor*));
+            MOCK_METHOD1(RegisterEntityAddedEventHandler, void(AZ::EntityAddedEvent::Handler&));
+            MOCK_METHOD1(RegisterEntityRemovedEventHandler, void(AZ::EntityRemovedEvent::Handler&));
+            MOCK_METHOD1(RegisterEntityActivatedEventHandler, void(AZ::EntityActivatedEvent::Handler&));
+            MOCK_METHOD1(RegisterEntityDeactivatedEventHandler, void(AZ::EntityDeactivatedEvent::Handler&));
+            MOCK_METHOD1(SignalEntityActivated, void(AZ::Entity*));
+            MOCK_METHOD1(SignalEntityDeactivated, void(AZ::Entity*));
+            MOCK_METHOD1(AddEntity, bool(AZ::Entity*));
+            MOCK_METHOD1(RemoveEntity, bool(AZ::Entity*));
+            MOCK_METHOD1(DeleteEntity, bool(const AZ::EntityId&));
+            MOCK_METHOD1(EnumerateEntities, void(const EntityCallback& /*callback*/));
+            MOCK_METHOD0(GetSerializeContext, AZ::SerializeContext*());
+            MOCK_METHOD0(GetBehaviorContext, AZ::BehaviorContext*());
+            MOCK_METHOD0(GetJsonRegistrationContext, AZ::JsonRegistrationContext*());
+            MOCK_CONST_METHOD0(GetEngineRoot, const char*());
+            MOCK_CONST_METHOD0(GetExecutableFolder, const char*());
+            MOCK_CONST_METHOD1(QueryApplicationType, void(AZ::ApplicationTypeQuery& /*appType*/));
+            MOCK_METHOD1(FindEntity, AZ::Entity*(const AZ::EntityId& inputId));
+    };
```

Comment:
[MockComponentApplication.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/UnitTest/MockComponentApplication.h) already implements the same mock.

This test could just include that header

Reply:
Yeah.  I'll also change MockComponentApplication.h so that those mock methods are not private / protected.  It makes no sense for mock methods to be private - it causes ON_CALL or EXPECT_CALL macros to not be able to be supplied from the outside which causes problems using the mock.